core/vm/instructions.go

```diff
@@ -260,7 +261,11 @@ func opAddress(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]
 func opBalance(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext) ([]byte, error) {
 	slot := scope.Stack.peek()
 	address := common.Address(slot.Bytes20())
-	slot.SetFromBig(interpreter.evm.StateDB.GetBalance(address))
+	balance, err := interpreter.evm.StateDB.GetBalance(address)
+	if err != nil {
```

Comment:
I added a comment, that should be easier to see for somebody looking at the code

Reply:
This comment doesn't make sense here. _Maybe_ if you wanted to add this as documentation next to _every_ op-code which behaves this way, but since the op-codes aren't documented (here), that would just be tedious and not very useful, IMO.

Even if you want to add code comments, I'd still argue this commit needs a commit message describing (at a high level) how this patch enforces sharding. So as we scan the commit history, we can see at a glance what this giant commit does without having to look at 25 files of code changes. It doesn't need to be anything crazy, just something simple like:
```
State db modifications to prevent out-of-scope state access

This commit enforces state sharding, by:
* checking addresses are in-scope before creating/reading/modifying state object
* returning out-of-scope errors so that EVM can capture in receipts
* etc, etc, etc
```
Obviously add whatever detail you want, but I should be able to know (at a high level) what a commit does without needing to read the code.